payments-ui-core/src/main/java/com/stripe/android/ui/core/forms/resources/LpmRepository.kt

```diff
@@ -96,8 +99,8 @@ class LpmRepository constructor(
     }
 
     /**
-     * This method will read the expected LPMs and their specs as two separate parameters.
-     * Any spec now found from the server will be read from disk json file.
+     * This method will read the [StripeIntent] and their specs as two separate parameters.
```

Comment:
How does this impact the ability to decouple StripeIntent from PaymentSheet? To me, this feels like a tighter coupling.

Reply:
We specifically need to know if it's a payment intent or a setup intent, as well as if a payment intent has setup_future_usage. @tillh-stripe can comment on what cashapp needs. The thing is, it would kinda explode if we completely decoupled it, because different LPMs have different requirements.

That being said. This is all internal, so we can refactor fairly easily.

Reply:
This comment is interesting though. It makes me think that our existing structure might not be ideal. We should decouple this mapping logic from `LpmRepository`.

If it was decoupled from `LpmRepository`, we don't care as much about the coupling mentioned.

Reply:
Talked a bunch offline. When we get to the decoupled future, `StripeIntent` will likely be the thing that implements a new interface that gives the data needed to `LpmRepository`. I think the best thing to do for today is to leave `StripeIntent`.